### PR TITLE
[ingest-geoip] update geoip to not include null-valued results from 

### PR DESCRIPTION
**Pull request by talevy**
for a better getting-started experience.

It is common for the local geolite database to have gaps in coverage,
this results in null-valued lat/lon fields. These fields conflict
with potential geo_point field mappings. By setting the parent
`location` field to null, the document can still be indexed with no
mapping issues.

### CODE REVIEW

**Comment by imotov** [comment]
@talevy the implementation looks good, but I don't quite understand the reasoning for having the field a `null` there instead of omitting the field completely. Is it so we can use null_value mapping and set a default value later on?

**Comment by talevy** [comment]
Updated to omit the null-value fields. If `geoip` is empty, then the field is not inserted at all

**Comment by talevy** [comment]
retest this please

**Comment by talevy** [comment]
retest this please

File: plugins/ingest-geoip/src/main/java/org/elasticsearch/ingest/geoip/GeoIpProcessor.java [hunk 1]
```diff
@@ -97,7 +97,9 @@ public void execute(IngestDocument ingestDocument) {
                 throw new ElasticsearchParseException("Unsupported database type [" + dbReader.getMetadata().getDatabaseType()
                         + "]", new IllegalStateException());
         }
-        ingestDocument.setFieldValue(targetField, geoData);
+        if (geoData.isEmpty() != true) {
```
Comment (imotov) [inline plugins/ingest-geoip/src/main/java/org/elasticsearch/ingest/geoip/GeoIpProcessor.java]: You probably meant geoData.isEmpty() == false

**Comment by imotov** [comment]
Left one really minor comment. LGTM.

File: plugins/ingest-geoip/src/main/java/org/elasticsearch/ingest/geoip/GeoIpProcessor.java (hunk 1 above)
Comment (talevy) [inline plugins/ingest-geoip/src/main/java/org/elasticsearch/ingest/geoip/GeoIpProcessor.java]: 😄 right

**Pull request merged**